src/adaptix/_internal/morphing/generic_provider.py

```diff
@@ -359,7 +360,9 @@ def _is_single_optional(self, norm: BaseNormType) -> bool:
         return len(norm.args) == 2 and None in [case.origin for case in norm.args]  # noqa: PLR2004
 
     def _is_class_origin(self, origin) -> bool:
-        return (origin is None or isinstance(origin, type)) and not is_subclass_soft(origin, collections.abc.Callable)
+        return (
+            origin is None or isinstance(origin, type) or origin is typing.Literal
+        ) and not is_subclass_soft(origin, collections.abc.Callable)
```

Comment:
``origin is typing.Literal`` should be moved to 

```python
forbidden_origins = [case.source for case in norm.args if not self._is_class_origin(case.origin) and case.origin != Literal]
```

Reply:
done